include/dynd/kernels/field_access_kernel.hpp

```diff
@@ -6,30 +6,62 @@
-    const uintptr_t *data_offsets;
+    const uintptr_t data_offset;
 
-    field_access_kernel(const uintptr_t *data_offsets) : data_offsets(data_offsets) {}
+    field_access_kernel(uintptr_t data_offset) : data_offset(data_offset) {}
 
-    void single(char *DYND_UNUSED(dst), char *const *src)
+    ~field_access_kernel() { get_child()->destroy(); }
+
+    void single(char *res, char *const *src)
+    {
+      char *const field_src[1] = {src[0] + data_offset};
+      get_child()->single(res, field_src);
+    }
+
+    static void resolve_dst_type(char *DYND_UNUSED(static_data), char *DYND_UNUSED(data), ndt::type &dst_tp,
+                                 intptr_t DYND_UNUSED(nsrc), const ndt::type *src_tp, intptr_t DYND_UNUSED(nkwd),
+                                 const array *kwds, const std::map<std::string, ndt::type> &DYND_UNUSED(tp_vars))
     {
-      std::cout << "x = " << *reinterpret_cast<int *>(src[0] + data_offsets[0]) << std::endl;
-      std::cout << "y = " << *reinterpret_cast<double *>(src[0] + data_offsets[1]) << std::endl;
+      dst_tp = get_field_type(src_tp, kwds);
     }
 
-    static void instantiate(char *DYND_UNUSED(static_data), char *DYND_UNUSED(data), kernel_builder *ckb,
-                            const ndt::type &DYND_UNUSED(dst_tp), const char *DYND_UNUSED(dst_arrmeta),
-                            intptr_t DYND_UNUSED(nsrc), const ndt::type *DYND_UNUSED(src_tp),
+    static void instantiate(char *DYND_UNUSED(static_data), char *data, kernel_builder *ckb, const ndt::type &dst_tp,
+                            const char *dst_arrmeta, intptr_t DYND_UNUSED(nsrc), const ndt::type *src_tp,
                             const char *const *src_arrmeta, kernel_request_t kernreq, intptr_t DYND_UNUSED(nkwd),
-                            const array *DYND_UNUSED(kwds),
-                            const std::map<std::string, ndt::type> &DYND_UNUSED(tp_vars))
+                            const array *kwds, const std::map<std::string, ndt::type> &tp_vars)
+    {
+      const uintptr_t data_offset = get_data_offset(src_tp, src_arrmeta, kwds);
+      const ndt::type field_type[1] = {get_field_type(src_tp, kwds)};
+      const nd::array field_value = nd::empty(field_type[0]);
+      const char *field_metadata[1] = {field_value->metadata()};
+
+      ckb->emplace_back<field_access_kernel>(kernreq, data_offset);
+
+      static const array error_mode(opt<assign_error_mode>());
+      assign::get()->instantiate(assign::get()->static_data(), data, ckb, dst_tp, dst_arrmeta, 1, field_type,
+                                 field_metadata, kernreq | kernel_request_data_only, 1, &error_mode, tp_vars);
+    };
+
+    static const ndt::type &get_field_type(const ndt::type *src_tp, const array *kwds)
+    {
+      const ndt::struct_type *s = src_tp->extended<ndt::struct_type>();
+      const std::string &name = kwds[0].as<std::string>();
+      uintptr_t index = s->get_field_index(name);
```

Comment:
Yeah, you're right, we need an error check like that. I'll add it in after I fix the other issues.